src/wms/src/main/java/org/geoserver/wms/map/GetMapKvpRequestReader.java

```diff
@@ -288,13 +301,27 @@ public GetMapRequest read(Object request, Map kvp, Map rawKvp) throws Exception
         if (layerParam != null) {
             List<String> layerNames = KvpUtils.readFlat(layerParam);
             requestedLayerInfos.addAll(parseLayers(layerNames, remoteOwsUrl, remoteOwsType));
+        } else if (citeCompliant && getMap.getSldBody() == null && getMap.getSld() == null) {
+            // The SLD extensions to WMS allow a request not to have layers, as long as a full SLD
+            // is specified either using &sld or &sld_body. The error must not be thrown in these
+            // conditions (which are probably not what INSPIRE had in mind, but nonetheless a OGC
+            // specification.
+            throw new ServiceException(
+                    "GetMap LAYERS parameter is mandatory if SLD nor SLD_BODY are not specified");
         }
 
         // raw styles parameter
         String stylesParam = (String) kvp.get("STYLES");
         List<String> styleNameList = new ArrayList<String>();
         if (stylesParam != null) {
             styleNameList.addAll(KvpUtils.readFlat(stylesParam));
+        } else if (citeCompliant && getMap.getSldBody() == null && getMap.getSld() == null) {
```

Comment:
Was about to merge but wondered about one thing... if the parameter is present but empty does everything still work even in CITE compliant mode?
From the spec:

> A client may request the default Style using a null value (as in "STYLES=").